frontend/cypress/integration/common/sidecar_injection.ts

```diff
@@ -0,0 +1,274 @@
+    cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+        metadata: {
+            labels: {
+                "istio-injection": null,
+                "istio.io/rev": null
+            }
+        }
+    });
+});
+
+Given('a namespace which has override configuration for automatic sidecar injection', function () {
+    this.targetNamespace = "default";
+    this.istioInjection = "enabled";
+
+    // Make sure that the target namespace has some override configuration
+    cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+        metadata: {
+            labels: {
+                "istio-injection": this.istioInjection,
+                "istio.io/rev": null
+            }
+        }
+    });
+});
+
+Given('the override configuration for sidecar injection is {string}', function (enabledOrDisabled) {
+    if (this.istioInjection !== enabledOrDisabled) {
+        cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+            metadata: {
+                labels: {
+                    "istio-injection": enabledOrDisabled,
+                    "istio.io/rev": null
+                }
+            }
+        });
+        this.istioInjection = enabledOrDisabled;
+    }
+});
+
+Given('a workload without a sidecar', function () {
+    this.targetNamespace = 'default';
+    this.targetWorkload = 'sleep';
+
+    // To achieve a workload without sidecar, we turn off injection in its namespace
+    // and also remove any override annotation for sidecar injection.
+    this.namespaceAutoInjectionEnabled = false;
+    this.workloadHasSidecar = false;
+    this.workloadHasAutoInjectionOverride = false;
+
+    // Make sure that injection in the namespace is turned off
+    cy.request('PATCH', '/api/namespaces/' + this.targetNamespace, {
+        metadata: {
+            labels: {
+                "istio-injection": null,
+                "istio.io/rev": null
+            }
+        }
+    });
+
+    // Make sure that the workload does not have override configuration
+    cy.request('PATCH', `/api/namespaces/${this.targetNamespace}/workloads/${this.targetWorkload}?type=Deployment`, {
+        spec: {
+            template: {
+                metadata: {
+                    labels: {
+                        "sidecar.istio.io/inject": null
+                    }
+                }
+            }
+        }
+    });
+
+    // Restart the workload to ensure changes are applied.
+    cy.exec('kubectl delete pod -n default -l app=sleep')
+});
+
+Given('a workload with a sidecar', function () {
+    this.targetNamespace = 'default';
+    this.targetWorkload = 'sleep';
+
+    // To achieve a workload with sidecar, we turn on injection in its namespace
+    // and also remove any override annotation for sidecar injection.
+    this.namespaceAutoInjectionEnabled = true;
+    this.workloadHasSidecar = true;
+    this.workloadHasAutoInjectionOverride = false;
```

Comment:
Do these need to be shared state or can they just be const globals for now?

Reply:
I was using shared state to prevent a large churn in some future PR is someday these steps need to become parametrized. Thus, you may simply add the parameters and assign to the state and all other steps can remain unchanged.

That said, I'm OK to change this to globals if you think it will be much better.

Reply:
I don't know if sharing state between the tests will make the tests easier or harder to maintain so can't recommend one way or another. It's definitely supported by the plugin. I'd say just leave it as it is for now.